File: src/main/java/com/jcfp/tallererp/model/Vehiculo.java
package com.jcfp.tallererp.model;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;

@Entity(name = "vehiculos")
@Getter
@Setter
public class Vehiculo {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @Column(unique = true, nullable = false)
    private String matricula;
    @ManyToOne
    @JoinColumn(name = "marca_id", nullable = false)
    private Marca marca;
    @ManyToOne
    @JoinColumn(name = "modelo_id", nullable = false)
    private Modelo modelo;
    private int matriculacion;
    @ManyToOne
    @JoinColumn(name = "cliente_id", nullable = false)
    private Cliente cliente;
}

File: frontend/src/types/entities/vehiculo.ts
export interface VehiculoPayload {
  matricula: string;
  marca: {
    id: string;
    nombre?: string;
  };
  modelo: {
    id: string;
    nombre?: string;
    marca_id?: string;
  };
  matriculacion: string;
  cliente: {
    id: string;
    nombre: string;
  };
}

export interface VehiculoEditData extends VehiculoPayload {
  id?: number;
}

export interface Marca {
  id: string;
  nombre: string;
}

export interface Vehiculo extends VehiculoPayload {
  id?: number;
}
